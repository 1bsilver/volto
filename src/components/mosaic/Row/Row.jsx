/**
 * Row component.
 * @module components/Row
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';

import { Column } from '../../../components';

/**
 * Row component class.
 * @function Row
 * @param {Object} props Component properties.
 * @param {Object} props.columns Column data.
 * @param {number} props.row Row index.
 * @param {func} props.selectTile Select tile method.
 * @param {func} props.setHoverd Set hovered tile method.
 * @param {func} props.setTileContent Set tile content method.
 * @returns {string} Markup of the row.
 */
const Row = ({ columns, row, selectTile, setHovered, setTileContent }) => (
  <Grid.Row>
    {columns.map((column, index) => (
      <Column
        row={row}
        column={index}
        width={column.width}
        tiles={column.tiles}
        selectTile={selectTile}
        setHovered={setHovered}
        setTileContent={setTileContent}
      />
    ))}
  </Grid.Row>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Row.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  row: PropTypes.number.isRequired,
  selectTile: PropTypes.func.isRequired,
  setHovered: PropTypes.func.isRequired,
  setTileContent: PropTypes.func.isRequired,
};

export default Row;
