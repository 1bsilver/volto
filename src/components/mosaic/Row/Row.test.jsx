import React from 'react';
import renderer from 'react-test-renderer';
import Row from './Row';

jest.mock('../Column/Column', () => jest.fn(() => <div />));

test('renders a row component', () => {
  const component = renderer.create(
    <Row
      columns={[
        {
          width: 2,
          tiles: [],
        },
      ]}
      row={0}
      selectTile={() => {}}
      deleteTile={() => {}}
      setHovered={() => {}}
      handleDrop={() => {}}
      setTileContent={() => {}}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
