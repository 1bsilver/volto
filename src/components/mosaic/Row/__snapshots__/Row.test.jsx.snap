// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a row component 1`] = `
<div
  className="undefined row"
>
  <div />
</div>
`;
